packages/libs/eda/src/lib/core/hooks/workspace.ts

```diff
@@ -160,7 +164,23 @@ export function useStudyEntities(filters?: Filter[]) {
                                   .rangeMax as number),
                         },
                       };
-                    else if (variable.type === 'date')
+                    else if (variable.type === 'date') {
+                      // recalculate bin width and units
+                      // to keep it simple let's keep the width at 1 and just try different units
+                      const binWidth = 1;
+                      const binUnits = (['year', 'month', 'week', 'day'].find(
+                        (unit) => {
+                          if (filterRange) {
+                            const diff = DateMath.diff(
+                              new Date(filterRange.min as string),
+                              new Date(filterRange.max as string),
+                              unit as DateMath.Unit
+                            );
+                            return diff >= 12;
```

Comment:
How did you arrive at 12?

Reply:
I left home at about 11:30 !

Reply:
But it was more "if there are more than 12 days, use weeks. If there are more than 12 weeks, use months, and so on."

Maybe it's too small. If there are 13 months, do we really want year bins?  Perhaps 30 would be better.

Reply:
Yeah, I'll go with 30. And leave a comment.